### DISCUSSION TITLE
More keyboard shortcuts

### DISCUSSION THREAD

**Discussion by diegogangl** [flxzt/rnote#234] [Ideas]
It would be great to have more keyboard shortcuts to have a smoother workflow.

**Tools:** : These could use the number row buttons from 1-6. Kinda of like how it works in FPSs (or in Blender's edit mode)
**Zoom**: Ctrl+middle click would be nice, either that or having the numpad `+` and `-` buttons
**Brush Size**: MyPaint handles this with two keys `d` and `f` to (in/de)crease the brush by one unit. Another option is shift+drag. Dragging outwards increase size, inwards decreases.
**Tool Settings**: A button to open the tools popup would be great too. This could be `tab`

**Comment by kotatsuyaki** [comment]
For those, like me, looking for a way to switch between tools using the number row buttons, I discovered that Ctrl+1 to Ctrl+6 already does that in the current implementation in [src/appwindow/actions.rs](https://github.com/flxzt/rnote/blob/3143fa35dbb0466a4afeaf1af938f3403049a590/crates/rnote-ui/src/appwindow/actions.rs#L776).

**Comment by Doublonmousse** [comment]
All shortcuts are listed in the Keyboard shortcut menu (`Ctrl + ?`) (just below `settings` in the menu)

![image](https://github.com/flxzt/rnote/assets/115779707/1180e68d-589d-4658-9da5-1b6c24ebe3e0)

![image](https://github.com/flxzt/rnote/assets/115779707/789c9662-1be0-4fc5-a184-ea0e6ceb24c8)

**Reply by Doublonmousse** [reply]
There's `ctrl+1` to 6 to change tools.
`Ctrl + +` and `ctrl + -` to zoom/dezoom are there (in addition to `ctrl + scroll`).

There's nothing for the brush size and tool settings (although I strongly disagree about using `Tab` for this, it's already used to switch focus between all UI elements and that's an accessibility feature).

**Reply by mahdihasan2006** [reply]
Currently missing `Ctrl + 0` for resetting zoom and shortcuts for colors.

`Ctrl + 0` is important because `Ctrl + +` and `Ctrl + -` can't reset zoom to 100%. Let's see how it works:

| Shortcut| Zoom Level|
|--------|--------|
| Default | 100% |
| `Ctrl + -` | 90% |
| `Ctrl + -` | 81% |
| `Ctrl + +` | 89% | 
| `Ctrl + +` | 98% |

**Reply by mahdihasan2006** [reply]
And shortcuts for colors will allow me to use the app better in Focus Mode. My laptop has a 1366x768px display resolution. So, the vertical space is very small compared to  horizontal space. And the app titlebar and two toolbars reduces it further. And these are bigger in low resolution compared to higher resolution. So, with these shortcuts, I will just use the app in focus mode as I mostly use pen and eraser tool and change colors.

**Comment by Fra0815** [comment]
How is one supposed to change these shortcut? Is there any shortcut to export a rnote file as pdf?

**Reply by hemantharunagiri-a11y** [reply]
Yeah I wish there was a way to change the shorcuts, cuz some keyword shortcuts are hard to do on had on the keyboard. Its difficult to quickly switch

**Comment by krzemyczek** [comment]
I miss shortcuts for colours too :c
